### PR TITLE
CI: Run common test on GHA

### PR DESCRIPTION
**Pull request by NobbZ**
Runs the erlang testsuite on CI.

### CODE REVIEW

File: .github/workflows/check-formatting.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
Comment (ErikSchierboom) [inline .github/workflows/check-formatting.yml:12]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/run-common-test.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
Comment (ErikSchierboom) [inline .github/workflows/run-common-test.yml:12]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/run-common-test.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - uses: actions/checkout@v3
```
Comment (ErikSchierboom) [inline .github/workflows/run-common-test.yml:14]:
```suggestion
      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
```

File: .github/workflows/check-formatting.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - uses: actions/checkout@v3
```
Comment (ErikSchierboom) [inline .github/workflows/check-formatting.yml:14]:
```suggestion
      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
```

File: .github/workflows/check-formatting.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: erlef/setup-beam@v1
```
Comment (ErikSchierboom) [inline .github/workflows/check-formatting.yml:15]:
```suggestion
      - uses: erlef/setup-beam@61e01a43a562a89bfc54c7f9a378ff67b03e4a21
```

File: .github/workflows/run-common-test.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: erlef/setup-beam@v1
```
Comment (ErikSchierboom) [inline .github/workflows/run-common-test.yml:15]:
```suggestion
      - uses: erlef/setup-beam@61e01a43a562a89bfc54c7f9a378ff67b03e4a21
```

**Review by ErikSchierboom — APPROVED** [review]
Nice work. Is this meant to replace #51?

File: .github/workflows/check-formatting.yml (hunk 4 above)
Comment (NobbZ) [reply .github/workflows/check-formatting.yml:14]: Oh, I hoped that the update bot would take care of that quickly after merge…

Are you able to suggest a tool that does that automatically, such that I do not have to look up the git rev ID manually?

**Comment by NobbZ** [comment]
@ErikSchierboom No, at best it is in addition.

Even though one of the suites runs docker, those tests are pure happy path and I am thinking about thinning it out. I had that originally to be able to easily see results locally. Now its just taking forever to run.

The main point of this for me is actually the unit-like tests.

And the tests have been there for quite a while, this just adds them to CI.

**Pull request merged**